File: String Permutations.cpp
    
    void solve(string S,vector<string>&s,int index){
        if(index>=S.size()){
            s.push_back(S);
            return;
        }
        
        for(int i=index;i<S.size();i++){
            swap(S[index],S[i]);
            solve(S,s,index+1);
            swap(S[i],S[index]);
        }
    }
    
    vector<string> permutation(string S)
    {
        vector<string>s;
        int index = 0;
        
        solve(S,s,index);
        sort(s.begin(),s.end());
        
        return s;
    }

File: String Permutations.java
class Solution
{    
    static ArrayList<String> list=new ArrayList<>();
    public ArrayList<String> permutation(String S)
    {   list.clear();
        perm(S,"");
        Collections.sort(list);
        return list;
    }
    static boolean perm(String str,String ans){
        if(str.length()==0)  return list.add(ans);
        for(int i=0;i<str.length();i++) perm(str.substring(0,i)+str.substring(i+1),ans+str.charAt(i)); 
        return true;
    }
}
